File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.java
class Solution {
    public int tribonacci(int n) {
             int t0=0,t1=1,t2=1;

            if (n==1  || n==2)return t2;

            int t_n=0;
            for (int i = 3; i < n+1; i++) {
                t_n=t0+t1+t2;
                t0=t1;
                t1=t2;
                t2=t_n;
            }
            return t_n;
    }
}

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n: int) -> int: 
        if n<=2: return 1 if n==1 or n==2 else 0
        ans=[0]*(n+1)
        ans[1]=ans[2]=1
        for i in range(3,n+1):
            ans[i]=ans[i-1]+ans[i-2]+ans[i-3]
             
        return ans[n]